airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/page_increment.py

```diff
@@ -19,15 +19,16 @@ class PageIncrement(PaginationStrategy):
         start_from_page (int): number of the initial page
     """
 
-    page_size: int
+    page_size: Optional[int]
     parameters: InitVar[Mapping[str, Any]]
     start_from_page: int = 0
 
     def __post_init__(self, parameters: Mapping[str, Any]):
         self._page = self.start_from_page
 
     def next_page_token(self, response: requests.Response, last_records: List[Mapping[str, Any]]) -> Optional[Any]:
-        if len(last_records) < self.page_size:
+        # Stop paginating when there are fewer records than the page size or the current page has no records
+        if (self.page_size and len(last_records) < self.page_size) or len(last_records) == 0:
```

Comment:
This is just our best effort attempt at stopping when the next page attempt has no more records included

I could also include this in the PageIncrement strategy, but right now `page_size` is not interpolated and we’ve haven’t seen a use case yet so I’d rather leave this alone and convert this to an interpolated value when there is a clear need.